processor/tailsamplingprocessor/processor.go

```diff
@@ -734,6 +790,71 @@ func (tsp *tailSamplingSpanProcessor) processTrace(id pcommon.TraceID, rss ptrac
 	}
 }
 
+// processEarlyDecisions tries to sample based on just the spans in a batch.
+func (tsp *tailSamplingSpanProcessor) processEarlyDecisions(id pcommon.TraceID, currentSpans ptrace.ResourceSpans, trace *samplingpolicy.TraceData) samplingpolicy.Decision {
+	var sampledPolicy *policy
+	finalDecision := samplingpolicy.Unspecified
+	notSampled := 0
+	// Check all policies before making a final decision.
+policyLoop:
+	for _, p := range tsp.policies {
+		earlyEval, ok := p.evaluator.(samplingpolicy.EarlyEvaluator)
+		if !ok {
+			continue
+		}
```

Comment:
we can avoid the cast by using `p.earlyEvaluator` I think?


```suggestion
		earlyEval := p.earlyEvaluator
		if earlyEval == nil {
			continue
		}
```